apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/monetary/service/AssetDeleteServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright © 2018-2020 Apollo Foundation
+ */
+
+package com.apollocurrency.aplwallet.apl.core.monetary.service;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.util.stream.Stream;
+
+import com.apollocurrency.aplwallet.apl.core.app.Transaction;
+import com.apollocurrency.aplwallet.apl.core.converter.IteratorToStreamConverter;
+import com.apollocurrency.aplwallet.apl.core.db.DbClause;
+import com.apollocurrency.aplwallet.apl.core.db.DbIterator;
+import com.apollocurrency.aplwallet.apl.core.db.service.BlockChainInfoService;
+import com.apollocurrency.aplwallet.apl.core.monetary.model.AssetDelete;
+import com.apollocurrency.aplwallet.apl.core.monetary.dao.AssetDeleteTable;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class AssetDeleteServiceImpl implements AssetDeleteService {
```

Comment:
I'd move those methods to the table level and create instead one unified AssetService

Reply:
That optimization is not highly necessary now.